### PR TITLE
nc news7

### PR DESCRIPTION
**Pull request by YoriB**
Hi need some help with 404 error handling. Seem to be going no where with the tests I ve done so far. Thanks.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+    const article_id = 3;
     return request(app)
-    .get('/api/comments/2/count')
+    .get(`/api/comments/${article_id}`)
     .expect(200)
-    .then(({body}) => {
-      console.log(body)
-    })
+    .then(({body}) => { 
+     for (let i = 0; i < body.length; i++) {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:121]: Why are you putting a length check in a for loop?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+})
+test('return a body of comments with the most recent first using created_at as the criteria', () => {
+  const article_id = 1;
+  return request(app)
+  .get(`/api/comments/${article_id}`)    
+    .then(({ body }) => {  
+       
+        expect(body).toBeSortedBy('created_at', {
+          descending: true,
+        });
```
Comment (alexdevdavis) [inline __tests__/app.test.js:151]: nice sorting test

File: __tests__/app.test.js [hunk 3]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+        });          
+    });
+});
+ 
+test('404 status response with a valid path but non existent article id', () => {
+  return request(app)
+  .get('/api/comments/55')
+  .expect(404)
+  .then(({body})=> {   
+    expect(body.msg).toEqual('Not found');
```
Comment (alexdevdavis) [inline __tests__/app.test.js:160]: can use `toBe` for primitive data types

File: __tests__/app.test.js [hunk 4]
```diff
@@ -110,17 +110,71 @@ test('400 status response with an  invalid article ID', () => {
...
+      expect(comment).toHaveProperty('votes');   
+    });
+  })  
+})
+test('return a body of comments with the most recent first using created_at as the criteria', () => {
+  const article_id = 1;
+  return request(app)
+  .get(`/api/comments/${article_id}`)    
+    .then(({ body }) => {  
+
```
Comment (alexdevdavis) [inline __tests__/app.test.js:148]: add a test to check that it returns an empty array for an article that exists, but has no comments yet

File: models/model.js [hunk 5]
```diff
@@ -37,22 +37,48 @@ const fetchArticlesById = (article_id) => {
...
+    FROM articles 
+    WHERE article_id = $1`, [article_id])
 
     .then((result) => {
       return result.rows[0];
     });
 };
 
-const fetchCommentsByArticleId = (article_id) => {
+// const fetchCommentsByArticleId = (article_id) => {
```
Comment (alexdevdavis) [inline models/model.js:49]: don't commit and push commented out code

File: test.sql [hunk 6]
```diff
@@ -1,27 +1,5 @@
 \c nc_news_test
```
Comment (alexdevdavis) [inline test.sql:1]: gitignore this file

File: test.txt [hunk 7]
```diff
@@ -1,17 +1 @@
 You are now connected to database "nc_news_test" as user "northcoders".
```
Comment (alexdevdavis) [inline test.txt:1]: and this one

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
Some additions and changes to make before moving on. 
- please don't ignore the feedback from before; I'm having to repeat myself

File: __tests__/app.test.js (hunk 1 above)
Comment (YoriB) [reply __tests__/app.test.js:121]: I am sure I deleted the for loop early on. It looks like I am not git committing the branches right as it seems  to be reverting to the earlier versions.
